python/foglamp/tasks/backup_restore/backup.py

```diff
@@ -0,0 +1,667 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017
+
+""" Backups the entire FogLAMP repository into a file in the local filesystem,
+it executes a full warm backup.
+
+The information about executed backups are stored into the Storage Layer.
+
+The parameters for the execution are retrieved from the configuration manager.
+It could work also without the configuration manager,
+retrieving the parameters for the execution from the local file 'backup_configuration_cache.json'.
+
+"""
+
+import time
+import sys
+import asyncio
+import os
+import json
+
+from foglamp.common.configuration_manager import ConfigurationManager
+from foglamp.common.storage_client import payload_builder
+from foglamp.common.process import FoglampProcess
+from foglamp.common import logger
+import logging
+
+import foglamp.tasks.backup_restore.lib as lib
+import foglamp.tasks.backup_restore.exceptions as exceptions
+
+__author__ = "Stefano Simonelli"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_MODULE_NAME = "foglamp_backup_module"
```

Comment:
I would suggest we indicate this is the Postgres backup module. See comment about support for different databases.

Reply:
changed to : foglamp_backup_postgres_module